jinja2/compiler.py

```diff
@@ -1314,7 +1314,12 @@ def visit_AssignBlock(self, node, frame):
         self.newline(node)
         self.visit(node.target, frame)
         self.write(' = (Markup if context.eval_ctx.autoescape '
-                   'else identity)(concat(%s))' % block_frame.buffer)
+                   'else identity)(')
+        if node.filter is not None:
+            self.visit_Filter(node.filter, block_frame)
+        else:
+            self.write('concat(%s)' % block_frame.buffer)
+        self.write(')')
```

Comment:
Not sure if this properly handles the new identifier tracking. This needs some tests what happens if you combine it with set and referring to variables in weird ways.